syft/cpe/by_source_then_specificity.go

```diff
@@ -0,0 +1,40 @@
+package cpe
+
+import "sort"
+
+type BySourceThenSpecificity []SourcedCPE
+
+func (b BySourceThenSpecificity) Len() int {
+	return len(b)
+}
+
+func (b BySourceThenSpecificity) Less(i, j int) bool {
```

Comment:
I'd also like some feedback on this sorting logic. Right now, I always have: prefer a CPE from the NVD dictionary if we have one, then one declared, then one syft generated, then any other source. Does that seem reasonable? I know in some places we want the best single CPE, so putting the most specific CPE from the NVD dictionary if we have one makes sense.

Reply:
synced on this last week, ultimately sounds good 👍